src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -1194,7 +1194,8 @@ trait Contexts { self: Analyzer =>
     }
 
     final def lookupCompanionOf(original: Symbol): Symbol = {
```

Comment:
> i.e., is it a term-owned symbol??

I think we also get into this code path for class-owned symbols where the class info is incomplete. So the check would need to be something like `owner.isTerm || !owner.hasCompleteInfo`